src/GitHub.Exports/Models/IBranch.cs

```diff
@@ -1,7 +1,15 @@
-﻿namespace GitHub.Models
+﻿using GitHub.Collections;
+using System;
+
+namespace GitHub.Models
 {
-    public interface IBranch
+    public interface IBranch : ICopyable<IBranch>,
+        IEquatable<IBranch>, IComparable<IBranch>
     {
+        string Id { get; }
         string Name { get; }
+        ISimpleRepositoryModel Repository { get; set; }
+        bool IsTracking { get; set; }
+        string DisplayName { get; set; }
```

Comment:
Could some of these properties be made readonly? It kinda worries me to have them all read/write as it can be difficult to tell how they can change. In what circumstances can they change?